src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -30,15 +30,21 @@
  * Authors: 2016 Pan Deng, Soumyajit De, Heiko Strathmann, Viktor Gal
  */
 
+#ifndef LINALG_NAMESPACE_H_
+#define LINALG_NAMESPACE_H_
+
 #include <shogun/mathematics/linalg/LinalgBackendBase.h>
 #include <shogun/mathematics/linalg/SGLinalg.h>
+#include <shogun/lib/Lock.h>
 
 namespace shogun
 {
 
 namespace linalg
 {
 
+static CLock m_gpu_transfer;
```

Comment:
is it a problem that `CLock` only has PTHREAD locking stuff?
We might need to add openmp, or is openmp locking implied by pthread?
http://stackoverflow.com/questions/2396430/how-to-use-lock-in-openmp

While openmp implementations are likely to be based on pthread, the specification does not talk about this --- I think we should better be safe than sorry
http://stackoverflow.com/questions/5661830/using-lock-in-openmp-and-pthreads